### PR TITLE
Only close seika connection if initialized

### LINKED ISSUE #13679: Revert configuration gives error
### Steps to reproduce:
**When you select 'No Braille' in the Braille device selection combo box and click the OK button, you will not be able to reproduce this.**

1. Launch NVDA2022.1Beta4;
2. Feel free to change a config, such as checking or unchecking an option in the Document Formatting panel, and then click the OK button;
3. Press Ctrl +NVDA+R to try to reverse this config.

### Actual behavior:
NVDA throws something errors and play an error sound.
I found by observing the logs that this should be a regression introduced by #13191.
Here are the key log snippets:
```
Loaded braille display driver seikantk, current display has 0 cells.
ERROR - queueHandler.flushQueue (21:34:56.030) - MainThread (9512):
Error in func resetConfiguration
Traceback (most recent call last):
  File "queueHandler.pyc", line 55, in flushQueue
  File "core.pyc", line 212, in resetConfiguration
  File "braille.pyc", line 2341, in terminate
  File "braille.pyc", line 1743, in terminate
  File "brailleDisplayDrivers\seikantk.pyc", line 191, in terminate
AttributeError: 'BrailleDisplayDriver' object has no attribute '_dev'
ERROR - eventHandler.executeEvent (21:34:57.220) - MainThread (9512):
error executing event: valueChange on <NVDAObjects.IAccessible.IAccessible object at 0x0E106090> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 285, in executeEvent
  File "eventHandler.pyc", line 99, in __init__
  File "eventHandler.pyc", line 108, in next
  File "NVDAObjects\IAccessible\__init__.pyc", line 1615, in event_valueChange
  File "NVDAObjects\__init__.pyc", line 1189, in event_valueChange
AttributeError: 'NoneType' object has no attribute 'handleUpdate'
```

### Expected behavior:
No Errors.

#### NVDA installed/portable/running from source:
installed and portable
#### NVDA version:
2022.1Beta4
#### Windows version:
Windows 10 21H2 (x64) build 19044.1682
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA2021.3.5 cannot reproduce this.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:

Closes part of #13679

### Summary of the issue:

It appears that when resetting a config, the seika notetaker driver can be terminated before it has been initialized.

### Description of how this pull request fixes the issue:

Only terminate the notetaker driver if it has been initialized.

### Testing strategy:

- [x] @cary-rowen to test a build of this PR

### Known issues with pull request:

Another issue is highlighted in #13679. While terminating/initializing during a config change, the braille.handler is attempted to be used when handling events on NVDA objects.
This is a more complex problem and needs a braille device to reproduce and test.

### Change log entries:
Fixes an unreleased regression

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by cary-rowen** [comment]
I tested this and still reproduce #13679

**Comment by cary-rowen** [comment]
[log.txt](https://github.com/nvaccess/nvda/files/8667751/log.txt)

**Comment by cary-rowen** [comment]
Hi, @seanbudd  
I tested the latest build and NVDA can catch this error:
```
ERROR - brailleDisplayDrivers.seikantk.BrailleDisplayDriver.terminate (10:50:45.576) - MainThread (10400):
Seika Notetaker driver not initialized
```

There seems to be another Error and I'm not quite sure how it happened:
```
RuntimeError: wrapped C/C++ object of type BoxSizer has been deleted
ERROR - unhandled exception (10:50:45.697) - MainThread (10400):
Traceback (most recent call last):
  File "wx\core.pyc", line 3407, in <lambda>
  File "gui\settingsDialogs.pyc", line 1457, in refreshGui
```

But my configuration can be reverse as expected.

Below is the full log:
[log.txt](https://github.com/nvaccess/nvda/files/8674703/log.txt)

**Comment by seanbudd** [comment]
Thanks @cary-rowen - I will change the log level of that message but this is as expected.

**Comment by XLTechie** [comment]
Rowen wrote:

> There seems to be another Error and I'm not quite sure how it happened:
> RuntimeError: wrapped C/C++ object of type BoxSizer has been deleted

I have also been seeing that on config resets lately. I suspect it is being 
caused by an add-on not (properly) cleaning up its config panel in terminate(), 
but haven't gone through the hassle of figuring out which one yet, if it is one 
and not core.

Unfortunately it doesn't happen with every reset, so has been difficult to pin 
down.

Next time please post a debug log.

**Comment by seanbudd** [comment]
> Rowen wrote:
> There seems to be another Error and I'm not quite sure how it happened: RuntimeError: wrapped C/C++ object of type BoxSizer has been deleted
> I have also been seeing that on config resets lately. I suspect it is being caused by an add-on not (properly) cleaning up its config panel in terminate(), but haven't gone through the hassle of figuring out which one yet, if it is one and not core. Unfortunately it doesn't happen with every reset, so has been difficult to pin down. Next time please post a debug log.

This is a problem with NVDA core - I have noticed this without add-ons installed.

**Comment by cary-rowen** [comment]
Hi, @seanbudd 

I tested it again and it seems to be working fine so far, Thanks for your efforts.
Please remember to put it on the 2022.1 milestone.

Thanks

**Comment by seanbudd** [comment]
> Hi, @seanbudd
> 
> I tested it again and it seems to be working fine so far, Thanks for your efforts. Please remember to put it on the 2022.1 milestone.
> 
> Thanks

Please see the comment on the issue
https://github.com/nvaccess/nvda/issues/13679#issuecomment-1124548979

File: source/brailleDisplayDrivers/seikantk.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def isSeikaBluetoothDeviceMatch(match: DeviceMatch) -> bool:
 
 
 class BrailleDisplayDriver(braille.BrailleDisplayDriver):
-	_dev: hwIo.IoBase
+	_dev: Optional[hwIo.IoBase] = None
```
Comment (feerrenrut) [inline source/brailleDisplayDrivers/seikantk.py:100]: Should this actually be made an instance variable?

File: source/braille.py [hunk 2]
```diff
@@ -2424,13 +2424,13 @@ def terminate(self):
...
-		super(BrailleDisplayDriver,self).terminate()
+		super().terminate()
 		# Clear the display.
 		try:
 			self.display([0] * self.numCells)
-		except:
+		except Exception as e:
 			# The display driver seems to be failing, but we're terminating anyway, so just ignore it.
-			pass
+			log.debugWarning(f"Display driver {self} is failing while terminating. {e}")
```
Comment (feerrenrut) [inline source/braille.py:2433]: Shouldn't this be logged at `error` level?

**Comment by seanbudd** [comment]
@cary-rowen - if we end up doing a 2022.1rc2 due to #13694, I think the fix for this issue should be included as well.

**Comment by cary-rowen** [comment]
@seanbudd 
I agree, I'll take a look too #13694

File: source/brailleDisplayDrivers/seikantk.py (hunk 1 above)
Comment (seanbudd) [reply source/brailleDisplayDrivers/seikantk.py]: this has been changed in a146544

File: source/braille.py (hunk 2 above)
Comment (seanbudd) [reply source/braille.py]: I agree, this has been changed in 941823b

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**